### ISSUE TITLE
SSH access concerns

### ISSUE THREAD

**Issue by shawark** [instaclustr/supportCenter#30]
SupportCenter assume the user can run from a machine with SSH access to all the nodes. However the assumption may fail in the following cases:

- default SSH may not work. For example our clusters use `go` to get to individual nodes and I cannot SSH from my laptop to any production servers. Therefore I cannot run either from admin box or my laptop (actually this is a problem for my testing. I may need to set up Cassandra on AWS and I have not done it yet).
- the script relies on remote nodes trust the machine's SSH key without putting a password
- the script assumes the user SSH having access to the folder or running `nodetool` command. Assuming some organization may require the user to elevate permission to perform above actions, e.g. `sudo` or `su -` with password
- the transparent gateway may not be set up for their cluster if the cluster only has private IPs
- customers need to change SSH user after SSH to the gateway server

For the above reasons, SSH access requirements may limit the usage of the script. I think it might be useful we can have a per node mode for the customer to run on an individual node and then copied out (however go script might not helpful since the customer may need to build for their various platform. Helping them to build go might be a non-trivial task).

**Comment by mrlika** [comment]
As we understand from the above some customers may require a custom tool or solution to gather logs and metrics from their cluster. 

We can provide the specification of the archive structure (i.e. directories and files required for each node) for such customers. And it is their responsibility to create and deliver a correct archive with the logs/metrics. They may build a custom solution that automatically does it in their environment.

**Comment by benbromhead** [comment]
@shawark for context: 

- SSH access in the current manner was asumed to be a easy common denominator. Most environments will support it in some form. 
- Also the binary doesn't execute ssh itself, but connects over the protocol. So it would likely work in our env (go wraps ssh in our internal env).
- ssh forwarding/bastion support is built into ssh (via the ProxyJump directive). Not sure if that is supported by supportCenter, but it's baked into ssh itself
- we made the assumption that network access to the nodes would be available... an offline mode (single node) would be useful for environments, but I think we would want to see an end customer request for it before working on it

**Comment by shawark** [comment]
Fair enough, that's why I only raise as a concern. The worst case we still have Instacollector.

**Comment by shawark** [comment]
I have a user case that the SSH user need to run all command with `sudo` to get configs/logs or run command like

```
du
df
cp
nodetool
```

It might worth reviewing this issue to see we can customise the command to add a prefix `sudo`

**Comment by jordanbraiuka** [comment]
I think the best solution here would be to have a general wrapper config option that can be passed in, that could add sudo, or other commands so that they can run. They could exec inside the container, or switch users, ultimately it gives us lots of flexibility. We could implement this as just a prefix in the first implementation. 

I think to start with we would probably want it as two _config options_: 
- A nodetool wrapper (Used for all the nodetool commands)
- A Config wrapper (Used for collecting log and configuration files)

That would be useful if the environment has different permissions for running nodetool commands, and the config files. 

We could consider making these commands config modifiable, but in the end I'd vote against it. We don't not want the config files to become extremely verbose.
